src/itdelatrisu/opsu/storyboard/TriggerEvent.java

```diff
@@ -0,0 +1,16 @@
+package itdelatrisu.opsu.storyboard;
+
+public enum TriggerEvent {
+	Passing, Failing, HitSoundFinish, HitSoundWhistle, HitSoundClap, None;
+	
+	public static TriggerEvent toTriggerEvent(String s) {

```

Comment:
You can use:

```
try {
    return TriggerEvent.valueOf(s);
} catch (IllegalArgumentException e) {
    return TriggerEvent.NONE;
}
```

Reply:
Relaying on try-catch logic is a code smell for me.

Reply:
[license.txt](https://github.com/itdelatrisu/opsu/files/7802025/license.txt)